packages/tangle/booker.go

```diff
@@ -549,28 +558,43 @@ func messageBranchUpdatedCaller(handler interface{}, params ...interface{}) {
 // MarkersManager is a Tangle component that takes care of managing the Markers which are used to infer structural
 // information about the Tangle in an efficient way.
 type MarkersManager struct {
-	tangle *Tangle
-
+	tangle         *Tangle
+	discardedNodes map[identity.ID]time.Time
 	*markers.Manager
 }
 
 // NewMarkersManager is the constructor of the MarkersManager.
 func NewMarkersManager(tangle *Tangle) *MarkersManager {
 	return &MarkersManager{
-		tangle:  tangle,
-		Manager: markers.NewManager(tangle.Options.Store, tangle.Options.CacheTimeProvider),
+		tangle:         tangle,
+		discardedNodes: make(map[identity.ID]time.Time),
+		Manager:        markers.NewManager(tangle.Options.Store, tangle.Options.CacheTimeProvider),
 	}
 }
 
 // InheritStructureDetails returns the structure Details of a Message that are derived from the StructureDetails of its
 // strong and like parents.
 func (m *MarkersManager) InheritStructureDetails(message *Message, sequenceAlias markers.SequenceAlias) (structureDetails *markers.StructureDetails) {
-	structureDetails, _ = m.Manager.InheritStructureDetails(m.structureDetailsOfStrongAndLikeParents(message), m.tangle.Options.IncreaseMarkersIndexCallback, sequenceAlias)
+	// customize the function, so that markers are not assigned to nodes whose messages might be dropped
+	// this is a temporary solution, that's why the function signature is not change to additionally accept messageID
+	structureDetails, _ = m.Manager.InheritStructureDetails(m.structureDetailsOfStrongAndLikeParents(message), func(sequenceID markers.SequenceID, currentHighestIndex markers.Index) bool {
+		nodeID := identity.NewID(message.IssuerPublicKey())
+		bufferUsedRatio := float64(m.tangle.Scheduler.BufferSize()) / float64(m.tangle.Scheduler.MaxBufferSize())
+		nodeQueueRatio := float64(m.tangle.Scheduler.NodeQueueSize(nodeID)) / float64(m.tangle.Scheduler.BufferSize())
+		if bufferUsedRatio > 0.01 && nodeQueueRatio > 0.1 {
+			return false
+		}
+		if discardTime, ok := m.discardedNodes[nodeID]; ok && time.Since(discardTime) < time.Minute {
+			return false
+		} else if ok && time.Since(discardTime) >= time.Minute {
```

Comment:
This method tells whether a new marker should be assigned to the message. What is done here, is we check whether any message from the same issuer has been discarded within the last minute. If there was such a message, then we don't want to use this message as a marker either, because there is a good chance that it might be problematic as well.